File: command/java/BuyStock.java
package Command;

// Nesnelere dönüştürdüğümüz her bir isteğe denk gelmektedir.
// Command (IOrderCommand) arayüzünü uygular.
// UML diyagramındaki ConcreteCommand sınıfına denk gelmektedir.
class BuyStock implements IOrderCommand {
    private StockManager _stockManager;

    public BuyStock(StockManager stockManager) {
        _stockManager = stockManager;
    }

    public void execute() {
        _stockManager.buy();
    }
}

File: command/kotlin/BuyStock.kt
package Command

// Nesnelere dönüştürdüğümüz her bir isteğe denk gelmektedir.
// Command (IOrderCommand) arayüzünü uygular.
// UML diyagramındaki ConcreteCommand sınıfına denk gelmektedir.
internal class BuyStock(private val _stockManager: StockManager) : IOrderCommand {

    override fun execute() {
        _stockManager.buy()
    }
}